### PR TITLE
Update Checkbox component

### LINKED ISSUE #2711: Update the Checkbox component
**AC:**

- [ ] Implement all the updates
- [ ] Create the `CheckboxGroup` component that can manage state of multiple checkboxes and return a single array of values. `<CheckboxGroup value={categoriesFilter} options={[]} onChange={(values) => void} />`

---

The goal of this task is to update the Radio button component based on the designs, docs, and changelog, which [can be found in Figma](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=7382%3A142509). 

>  **Changelog (from Figma)**
> - Updated fill, border and text colors to use color design tokens,
> - Updated transition animations to use animation design tokens,
> - Rearranged and renamed component properties,
> - Added Styling property with Error variants (for selected States),
> - Added “Checkbox group” component for easier and more consistent vertical stacking,
> - Added documentation along with content/usage guidelines,
> - Added Focus state, which inherits its styling from the Hover state.

image

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #2711

**Design**
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=7382%3A142509

### CODE REVIEW

**Review by toiletgranny — CHANGES_REQUESTED** [review]
Great, thanks! All good, except for:
1. The helper text is still `var(--color-core-neutral-300)`, should be `var(--color-text)`,
2. This changelog item seems to not be implemented: "Added Focus state, which inherits its styling from the Hover state".  The goal here is to do the same thing that @drillprop has done with the Switch component recently. Ideally, we use the the `:focus-visible`, instead of `:focus`, if that's okay with you.

**Review by drillprop — CHANGES_REQUESTED** [review]
Same issue
https://github.com/Joystream/atlas/pull/2786#pullrequestreview-996212641
Basically, when someone clicks input with a mouse we don't want to show regular focus state.

**Review by toiletgranny — APPROVED** [review]

File: packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts [hunk 1]
```diff
@@ -1,82 +1,63 @@
 import { css } from '@emotion/react'
-import styled from '@emotion/styled/'
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts:2]: **FYI** this caused me a lot of trouble. I had weird errors from babel in storybook and it turned out this was the reason...

**Review by toiletgranny — APPROVED** [review]

File: packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts [hunk 2]
```diff
@@ -1,82 +1,63 @@
...
-
-export const Input = styled.input`
+const hoverUncheckedStyles = css`
+  + ${Checkmark} {
+    ::before {
+      background-color: ${cVar('colorBackgroundStrongAlpha')};
+    }
+  }
+`
+export const Input = styled.input<InputProps>`
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts]: Let's rename this component. Maybe `CheckboxInput`?

File: packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts [hunk 3]
```diff
@@ -1,82 +1,63 @@
...
-        props.selected &&
-          css`
-            background-color: ${cVar('colorCoreRed400')};
-          `,
-      ]
-    : null
-export type CheckboxStateProps = {
-  selected: boolean
-  disabled: boolean
+type InputProps = {
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts]: same as above

File: packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts [hunk 4]
```diff
@@ -85,15 +66,93 @@ export const Input = styled.input`
   height: 100%;
   position: absolute;
   cursor: inherit;
-`
+  z-index: 999;
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts]: same comment https://github.com/Joystream/atlas/pull/2786#discussion_r893522142

File: packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts [hunk 5]
```diff
@@ -1,82 +1,63 @@
...
   color: ${cVar('colorCoreNeutral300')};
-  transition: background ${transitions.timings.loading} ${transitions.easing};
+`
+
+export const Checkmark = styled.div<{ error: boolean }>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 16px;
+  height: 16px;
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/Checkbox/Checkbox.styles.ts]: we could use `square` here, right?

File: packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    className: { table: { disable: true } },
+    selectedValue: { table: { disable: true } },
+    options: { table: { disable: true } },
+    value: { table: { disable: true } },
+    onChange: { table: { disable: true } },
+    name: { table: { disable: true } },
+  },
+  args: {
+    label: 'Hello there',
+    helperText: 't100',
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.stories.tsx]: Can we rename `helperText`  to `caption`? That's what it's called in Figma.  I think we should use the same names as much as we can.

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      label: args.label,
+      helperText: args.helperText,
+    },
+    {
+      label: args.label,
+      helperText: args.helperText,
+    },
+    {
+      label: args.label,
+      helperText: args.helperText,
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.stories.tsx]: You still used `helperText` here. Why does typescript think that everything is fine here?

**Review by drillprop — CHANGES_REQUESTED** [review]

**Comment by kdembler** [comment]
Needs to target `dev` (and optionally rebase)

File: packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+export const CheckboxGroup: React.FC<CheckboxGroupProps> = ({ options, value, onChange, name, disabled, error }) => {
+  return (
+    <Wrapper>
+      {options.map((option, idx) => (
+        <Checkbox
+          disabled={disabled}
+          error={error}
+          name={name}
+          key={`radio-button-group-${idx}`}
+          label="test"
```
Comment (eldiegod) [inline packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.tsx]: This is wrong right?

File: packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import styled from '@emotion/styled'
+import React from 'react'
+
+import { Checkbox, CheckboxProps } from '@/components/_inputs/Checkbox'
+import { sizes } from '@/styles'
+
+export type CheckboxGroupProps = {
+  options: CheckboxProps[]
+  onChange?: (id: number, event: React.ChangeEvent<HTMLInputElement>) => void
+  value: number[]
```
Comment (eldiegod) [inline packages/atlas/src/components/_inputs/CheckboxGroup/CheckboxGroup.tsx]: I don't think `value` is a very clear name when the type is `number[]` on a checkboxGroup, what about something on the lines of `checkedIds`

**Review by eldiegod — CHANGES_REQUESTED** [review]
Looks pretty good just found a couple details

**Review by drillprop — APPROVED** [review]
as for me, looks good

**Review by eldiegod — APPROVED** [review]

**Pull request merged**